pydantic/fields.py

```diff
@@ -1029,3 +1117,37 @@ class DeferredType:
     """
     Used to postpone field preparation, while creating recursive generic models.
     """
+
+
+def _get_discriminator_values(tp: Any, discriminator_key: str) -> Tuple[str, ...]:
```

Comment:
this file is already very long, could we move this logic somewhere else like `utils.py`?